cluster-autoscaler/metrics/liveness.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+)
+
+type HealthCheck struct {
+	lastActivity time.Time
+	mutex        *sync.Mutex
+}
+
+const (
+	timeout = time.Minute * 10
+)
+
+var (
```

Comment:
Not sure if we should put it into AutoscalingContext as it's not easily available from main() function (where we need to create handler) and it can get re-created by dynamic autoscaler. We can just put a pointer in AutoscalingOpts, but that would require passing it all the way from main. I think we can do all the required checks in run() function in main.go instead.

That way we can create the variable in main() and just pass it to run().